client/app/bundles/course/user-invitations/components/tables/UserInvitationsTable.tsx

```diff
@@ -0,0 +1,244 @@
+          const invitation = invitations[dataIndex];
+          return (
+            <Typography key={`role-${invitation.id}`} variant="body2">
+              {
+                sharedConstants.USER_ROLES.find(
+                  (role) => role.value === invitation.role,
+                )?.label
+              }
+            </Typography>
+          );
+        },
+      },
+    },
+    {
+      name: 'phantom',
+      label: intl.formatMessage(tableTranslations.phantom),
+      options: {
+        alignCenter: false,
+        customBodyRenderLite: (dataIndex: number): JSX.Element => {
+          const invitation = invitations[dataIndex];
+          return (
+            <Typography key={`phantom-${invitation.id}`} variant="body2">
+              {invitation.phantom ? 'Yes' : 'No'}
+            </Typography>
+          );
+        },
+      },
+    },
+    {
+      name: 'invitationCode',
+      label: intl.formatMessage(tableTranslations.invitationCode),
+      options: {
+        alignCenter: false,
+        customBodyRenderLite: (dataIndex: number): JSX.Element => {
+          const invitation = invitations[dataIndex];
+          return (
+            <Typography key={`invitationCode-${invitation.id}`} variant="body2">
+              {invitation.invitationKey}
+            </Typography>
+          );
+        },
+      },
+    },
+  ];
+
+  if (pendingInvitations) {
+    columns.push({
+      name: 'sentAt',
+      label: intl.formatMessage(tableTranslations.invitationSentAt),
+      options: {
+        alignCenter: false,
+        customBodyRenderLite: (dataIndex: number): JSX.Element => {
+          const invitation = invitations[dataIndex];
+          return (
+            <Typography key={invitation.id} variant="body2">
+              {invitation.sentAt}
+            </Typography>
+          );
+        },
+      },
+    });
+  }
+
+  if (acceptedInvitations) {
+    columns.push({
+      name: 'acceptedAt',
+      label: intl.formatMessage(tableTranslations.invitationAcceptedAt),
+      options: {
+        alignCenter: false,
+        customBodyRenderLite: (dataIndex: number): JSX.Element => {
+          const invitation = invitations[dataIndex];
+          return (
+            <Typography key={invitation.id} variant="body2">
+              {invitation.confirmedAt}
+            </Typography>
+          );
+        },
+      },
+    });
+  }
+
+  if (renderRowActionComponent) {
+    columns.push({
+      name: 'actions',
+      label: intl.formatMessage(tableTranslations.actions),
+      options: {
+        empty: true,
+        sort: false,
+        alignCenter: true,
+        customBodyRender: (_value, tableMeta): JSX.Element => {
+          const rowData = tableMeta.currentTableData[tableMeta.rowIndex];
+          const user = rebuildObjectFromRow(columns, rowData);
+          const actionComponent = renderRowActionComponent(user);
+          return actionComponent;
+        },
```

Comment:
This idea is also repeated in the other tables (EnrolRequestsTable, ManageUsersTable) but it could be one source of performance issue. On every row's action column, we need to pass the row's data into the action component. It will eagerly map through the entire table and fetch each row's data, then create the action component. Not sure if there's a way to only load this when the row's Action Component is triggered.

Reply:
You can add a memo to the component to be rendered. (eg UserManagementButtons)